frontend/src/_helpers/appUtils.js

```diff
@@ -82,38 +82,126 @@ export function addToLocalStorage(object) {
 export function getDataFromLocalStorage(key) {
   return localStorage.getItem(key);
 }
+async function exceutePycode(payload, code, currentState, query, mode) {
+  const subpath = window?.public_config?.SUB_PATH ?? '';
+  const assetPath = urlJoin(window.location.origin, subpath, '/assets');
+  const pyodide = await window.loadPyodide({ indexURL: `${assetPath}/py` });
+
+  const evaluatePython = async (pyodide) => {
+    let result = {};
+    try {
+      const _code = code.replace('return', '');
+      const _currentState = JSON.stringify(currentState);
+      let execFunction = await pyodide.runPython(`
+        from pyodide.ffi import to_js
+        import json
+        def exec_code(payload):
+          data = json.loads(payload)
+          currentState = json.loads('${_currentState}')
+          components = currentState['components']
+          queries = currentState['queries']
+          globals = currentState['globals']
+          variables = currentState['variables']
+          client = currentState['client']
+          server = currentState['server']
+          code_to_execute = ${_code}
+          try:
+            res = to_js(json.dumps(code_to_execute))
+            # convert dictioanry to js object
+            return res
+          except Exception as e:
+            print(e)
+            return {"error": str(e)}
+            
+        exec_code
+    `);
+      const _data = JSON.stringify(payload);
+      result = execFunction(_data);
```

Comment:
I'm getting an error saying `execFunction` is not a function on this line

Reply:
as it is scoped to the query, we have to return data or a primitive type